tests/rados/test_bug_fixes.py

```diff
@@ -298,3 +302,137 @@ def run(ceph_cluster, **kw):
             "Verification of automatic removal of Slow OSD heartbeat warning has been completed "
         )
         return 0
+
+    if config.get("lower_bluefs_shared_alloc_size"):
+        doc = (
+            "\n# CEPH-83591092"
+            "\n Bugzilla trackers:"
+            "\n\t- Pacific: 2264054"
+            "\n\t- Quincy: 2264053"
+            "\n\t- Reef: 2264052 & 2260306"
+            "\n Verify OSDs do not crash terminally when custom 'bluefs_shared_alloc_size' "
+            "is in use and it's below 'bluestore_min_alloc_size' persistent for BlueStore on deployment"
+            "\n\t 1. Deploy a cluster with no default values of 'bluestore_min_alloc_size' and "
+            "'bluefs_shared_alloc_size'"
+            "\n\t 2. Fetch the current active value of 'bluestore_min_alloc_size'"
+            "\n\t 3. Choose one OSD from each OSD node at random"
+            "\n\t 4. Set the value of 'bluefs_shared_alloc_size' to half of 'bluestore_min_alloc_size' "
+            "for each chosen OSD"
+            "\n\t 5. Restart Orch OSD service"
+            "\n\t 6. Fill the cluster till 70%"
+            "\n\t 7. Ensure restart of Orch OSD service is successful"
+        )
+
+        log.info(doc)
+        log.info(
+            "Running test to verify OSD resiliency when 'bluefs_shared_alloc_size'"
+            " is below 'bluestore_min_alloc_size'"
+        )
+
+        try:
+            _pool_name = "test-bluefs-shared"
+
+            # get the list of OSDs on the cluster
+            osd_list = rados_obj.run_ceph_command(cmd="ceph osd ls")
+            log.debug(f"List of active OSDs: \n{osd_list}")
+
+            # check the current value of 'bluestore_min_alloc_size'
+            min_alloc_size = int(
+                mon_obj.show_config(
+                    daemon="osd", id=osd_list[0], param="bluestore_min_alloc_size_hdd"
+                )
+            )
+            log.info(
+                f"Current persistent value of 'bluestore_min_alloc_size_hdd' is {min_alloc_size}"
+            )
+            min_alloc_size_ssd = int(
+                mon_obj.show_config(
+                    daemon="osd", id=osd_list[0], param="bluestore_min_alloc_size_ssd"
+                )
+            )
+            log.info(
+                f"Current persistent value of 'bluestore_min_alloc_size_ssd' is {min_alloc_size}"
+            )
+
+            # cherry-pick one OSD ID from each osd node
+            test_osd_list = [
+                rados_obj.collect_osd_daemon_ids(osd_node=node)[0] for node in osd_nodes
+            ]
+
+            log.info(
+                f"OSDs for which bluefs_shared_alloc_size will be modified: {test_osd_list}"
+            )
+            shared_alloc_size = int(min_alloc_size / 2)
```

Comment:
When the suite is run on BM systems with SSD, we are not currently considering the fetched value "min_alloc_size_ssd" anywhere? Should the test have intelligence to choose from HDD vs SSD value to determine the shared_alloc_size?

Reply:
Okay, currently, the default value of both `min_alloc_size_hdd` and `min_alloc_size_ssd` is **4096** and changing these values is not advisable, so the probability of having custom value of `min_alloc_size_ssd` which differs from `min_alloc_size_hdd` is very very low, because these values are set during deployment of OSDs and cannot be changed unless OSD is destroyed and redeployed. Additionally, regardless of whether the OSD has a HDD or SSD backend, the `shared_alloc_size` calculated for it will be 2048, so it will always satisfy the condition we are trying to test:  `bluefs_shared_alloc_size_[hdd,ssd] ` **<** `bluestore_min_alloc_size`